### ISSUE TITLE
Cannot build mod

### ISSUE THREAD

**Issue by Lukasz-Pe** [Ch4nKyy/BG3WASD#39]
Hello,

i was trying to build your mod, but got errors despite following your instructions. CMake wants tomlplusplusConfig.cmake.

Here are last lines of output with error:
```
-- Searching for DKUtil using file include/DKUtil/Logger.hpp
-- Found DKUtil in D:/Soft/BG3WASD/extern/DKUtil; adding
CMake Error at C:/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  Could not find a package configuration file provided by "tomlplusplus" with
  any of the following names:

    tomlplusplusConfig.cmake
    tomlplusplus-config.cmake

  Add the installation prefix of "tomlplusplus" to CMAKE_PREFIX_PATH or set
  "tomlplusplus_DIR" to a directory containing one of the above files.  If
  "tomlplusplus" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  extern/DKUtil/CMakeLists.txt:34 (find_package)
```

Could you help me resolve the issue?

**Comment by Ch4nKyy** [comment]
Hm is your VCPKG_ROOT path correct?
tomlplusplus is listed as a dependency in vcpkg.json, so it should be installed automatically during the cmake build.

**Comment by Lukasz-Pe** [comment]
I hope that yes.

in cmd.exe>
```
echo %VCPKG_ROOT%
```
gives correct path:
```
c:\vcpkg\
```
and during copilation there are no errors regarding problems with finding vcpkg.exe.
I have installed it from link in readme.

**Comment by toriningen** [comment]
I had the exactly same issue. The reason is that [DKUtil](https://github.com/gottyduke/DKUtil/) which BG3WASD depends on includes tomlplusplus incorrectly - perhaps tomlplusplus itself has updated, and since vkpkg has no version pinning it broke dependent builds.

There are two ways to fix it.

The easy way is to update the DKUtil submodule to point to the latest (for now) commit:
```powershell
git submodule set-branch --branch 63501cad99be025f08e79dc07bc9f060081bf916 -- extern/DKUtil
git submodule update --remote
```
Newer DKUtil includes tomlplusplus in-tree, so CMake will have no issues searching for it, and it seems to be the preferred solution.

The hard way is to patch DKUtil and BG3WASD accordingly to include "pkgconf" as vcpkg dependency, and fix the find_package() calls to use PkgConfig instead. To do this, look at [this commit to DKUtil](https://github.com/gottyduke/DKUtil/commit/10d4623f443a310184edd5d6a6dad6c9ddcda876) and [this commit to BG3WASD](https://github.com/Ch4nKyy/BG3WASD/commit/51004f44bdae908ca5bc7410a775b8ead6d3b2eb). I am not going to create a pull request for these changes, because DKUtil has moved on since and my changes are obsolete, and for BG3WASD there is better option - update the submodule. But if you need a quick fix, I guess that would do.

**Comment by Ch4nKyy** [comment]
Thanks a lot! I merged your PR.
That reminded me that I still had local changes in DKUtil that fix a random deadlock when using ScriptExtender.
If you run into that, here is the PR/commit to fix it:
https://github.com/gottyduke/DKUtil/pull/20

**Comment by toriningen** [comment]
> That reminded me that I still had local changes in DKUtil that fix a random deadlock when using ScriptExtender. If you run into that, here is the PR/commit to fix it: [gottyduke/DKUtil#20](https://github.com/gottyduke/DKUtil/pull/20)

Thank you, it seems that I've ran into this issue as well - BG3 just freezes upon starting before creating any windows, but I didn't have the time to debug it properly yet. I hope your patch is for the issue I'm thinking about - but if not, I guess I'll make another one :)

**Comment by Ch4nKyy** [comment]
When you build the mod in debug mode, the game will wait until you attach the debugger.
https://github.com/Ch4nKyy/BG3WASD/blob/main/src/main.cpp#L41

If you knew this already, then the issue was probably the fix that was just merged.
